src/plugins/controls/public/options_list/components/options_list_popover.tsx

```diff
@@ -52,12 +53,13 @@ export const OptionsListPopover = ({
   return (
     <div
       id={`control-popover-${id}`}
+      className={`optionsList__popover`}
```

Comment:
@Heenawter not it was not necessary, since there already was `scss` file so I just added the [style in that](https://github.com/elastic/kibana/pull/149121/files/c9400bad795b39d107dfe9a4591892b515a6d6fc#diff-1950fe08be7a0d7e43c25b1ff33e0b0bc8c8325297d9897a07b3e7fc8ec02a7cR91-R94).

Reply:
I guess I'm still a bit confused. Right below this, we have:
```typescript
style={{ width: width > 300 ? width : undefined }}
```
I think if we are going the route of styling through a class, then we should remove this - styling through both a `scss` and a `style` attribute for a single component can cause two sources of truth, which is hard to follow imo.

Reply:
You are right and it was my fault I did not pay attention to the style prop. Previously, if `width` < `300`, then we were setting width as `undefined`, not `300`. So now I have changed  `min-width` to 300px. Let me know if it make sense now.

Reply:
I don't think the new `min-width` is working quite as expected, either. Here's what happens when you shrink the window to be less than `300px` wide with your current code:

![Screenshot 2023-01-23 at 9 53 49 AM](https://user-images.githubusercontent.com/8698078/214100623-1ff093e4-1743-4731-84c0-6345efd95793.png)


Notice that the white part of the popover shrinks, but neither the inner contents nor the footer do - so you end up with a weird "overhang" effect. Whereas this is what happens with the original sizing code

![image](https://user-images.githubusercontent.com/8698078/214100690-1d6d0cf3-9346-41a8-91be-8e9f194425f8.png)


The sizing of the popovers is finicky! So we don't actually technically have a minimum size - we are allowing the **popover** to decide the minimum size, hence setting the width to `undefined`. If you can figure out a better way to handle this, I'd be down - but in its current state, not a fan of what happens when you set `min-width` like that.

Reply:
Thanks @Heenawter  for diving into this problem. However, I think it might not be that big of a problem. Please bear with me for the explanation.

It looks like there are 4 ways we can go about this problem:

1. Setting `width` to  `undefined` if width < 300 i.e. `width: width > 300 ? width : undefined  `
   - This is the original definition and it makes the popover look as below if we remove `ActionBar` so I guess we will need to have `width` for sure.
   - ![image](https://user-images.githubusercontent.com/7485038/214269266-b186ad8f-899a-4967-afcb-7d94bcaa88b3.png)


2. Not setting `min-width` and setting width to  the size of option list. Issue with this is that, option list can get very small in some cases. See below: 
   - ![image](https://user-images.githubusercontent.com/7485038/214269928-950eb6e1-043e-4547-a927-1ed825c7e628.png)

3. Because of these 2 reasons, I think setting `min-width` becomes important. Now one option is to set the min-width of the popover content as done above i.e `min-width : 300px`
   - I think this works well in our case because `screen` width will never( rarely?) be `<300` causing the issue that you just highlighted.
   - If you check [EuiTour component](https://eui.elastic.co/#/display/tour#managed-state-via-euitour-render-prop-component), it also suffers from the same problem. I will raise an issue with them.


4. Solution to the problem in the 3rd approach is passing `panelStyle` to EuiPopover. 
    - This solves above problem but it `may` give rise to other [positioning problems mentioned here](https://github.com/elastic/eui/pull/5897).   
    - Although I have not encountered any issue in our case so I am good with this approach as well but I prefer 3rd approach more.

Reply:
Ahh, thank you for the screenshots! That helps a lot to understand why you added that - makes total sense. I did not consider how the width would be impacted when the action bar was not present. I think that, given what you've outlined, sticking with your current `min-width` approach seems totally fair!

Reply:
🙏 Thank you very much.